homeassistant/components/bzutech/sensor.py

```diff
@@ -0,0 +1,373 @@
+    "EP111": ["SHT20-TMP", "SHT20-HUM", "BH1750-LUM", "SHT30-TMP", "SHT30-HUM"],
+    "EP121": ["SHT20-TMP", "SHT20-HUM", "BH1750-LUM", "DOOR-DOR"],
+    "EP200": [
+        "SHT20-TMP",
+        "SHT20-HUM",
+        "SGP30-VOC",
+        "SGP30-CO2",
+        "SPS30-M01",
+        "SPS30-M25",
+        "SPS30-M40",
+        "SPS30-M10",
+    ],
+    "EP300": ["RTZSBZ-SND"],
+    "EP400": [
+        "ADS7878-VRA",
+        "ADS7878-VRB",
+        "ADS7878-VRC",
+        "ADS7878-CRA",
+        "ADS7878-CRB",
+        "ADS7878-CRC",
+        "ADS7878-CRN",
+        "ADS7878-APA",
+        "ADS7878-APB",
+        "ADS7878-APC",
+        "ADS7878-PPA",
+        "ADS7878-PPB",
+        "ADS7878-PPC",
+        "ADS7878-RPA",
+        "ADS7878-RPB",
+        "ADS7878-RPC",
+        "ADS7878-ACA",
+        "ADS7878-BCA",
+        "ADS7878-CCA",
+    ],
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Do entry Setup."""
+    bzu_api = hass.data[DOMAIN][entry.entry_id]
+    sensors = []
+
+    for sensor in EndpointsSensors[entry.data[CONF_ENDPOINT]]:
+        sense = sensor + "-" + entry.data[CONF_SENSORPORT]
+        for description in SENSOR_TYPE:
+            if description.key == sense.split("-")[1]:
+                sensors.append(
+                    BzuEntity(bzu_api, sense, entry, description=description)
+                )
+
+    async_add_entities(sensors, update_before_add=True)
+
+
+class BzuEntity(SensorEntity):
+    """Setup sensor entity."""
+
+    def __init__(
+        self,
+        api: BzuTech,
+        sensorname,
+        entry: ConfigEntry,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Do Sensor configuration."""
+
+        self._attr_unique_id = (
+            entry.data[CONF_CHIPID]
+            + sensorname.split("-")[1]
+            + entry.data[CONF_SENSORPORT]
+        )
+        self.chipid = entry.data[CONF_CHIPID]
+        self.sensorname = sensorname
+        self.api: BzuTech = api
+        self.api.dispositivos[self.chipid].sensores.keys()
+        self.entity_description = description
+        self._attr_translation_key = description.key
+        self._attr_is_on = True
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, "ESP-" + self.chipid)},
+            suggested_area="Room",
+            name="Gateway " + self.chipid,
+            entry_type=DeviceEntryType("service"),
+            manufacturer="Bzu Tech",
+            hw_version="1.0",
+            model="ESP-" + self.chipid,
+            serial_number=self.chipid + "P" + entry.data[CONF_SENSORPORT],
+            sw_version="1.0",
+        )
+
+    async def async_update(self) -> None:
+        """Update entity values."""
+        try:
+            self._attr_native_value = await self.api.get_reading(
+                str(self.chipid), self.sensorname
+            )
```

Comment:
Why don't we use a DataUpdateCoordinator?

Reply:
couldn't get the Coordinator to update every entity of the device.

Reply:
Did you make sure the entities subscribed to updates from the coordinator?

Reply:
@enioo22 please follow up on this 👍

Reply:
Oh, I see, in that case it's fine to not use a coordinator 👍 
Maybe add a comment in the sensor class explaining there's no endpoint to get multiple sensor readings.